apps/events/models.py

```diff
@@ -169,7 +173,7 @@ def can_display(self, user):
             return True
         if not user:
             return False
-        return restriction.has_access(user)
+        return restriction.has_access(user) if self.visible else False
```

Comment:
Would this not still return `True` if the event has no GroupRestriction?

Reply:
Yes, because of `if not restriction: return True` earlier in the function

Reply:
If I'm not completely wrong `visible` should have higher precedence than a GroupRestriction?

In this case, all events will be visible if they don't have a GroupRestriction, which is almost all events.

Reply:
Yes, you are correct, fixed it now.